replication/binlogsyncer.go

```diff
@@ -830,24 +815,37 @@ func (b *BinlogSyncer) parseEvent(s *BinlogStreamer, data []byte) error {
 		if b.prevGset == nil {
 			break
 		}
+		if b.currGset == nil {
+			b.currGset = b.prevGset.Clone()
+		}
 		u, _ := uuid.FromBytes(event.SID)
-		err := advanceCurrentGtidSet(u, event.GNO, 0, 0, 0)
-		if err != nil {
-			return errors.Trace(err)
+		b.currGset.(*MysqlGTIDSet).AddGTID(u, event.GNO)
+		if b.prevMySQLGTIDEvent != nil {
+			u, _ = uuid.FromBytes(b.prevMySQLGTIDEvent.SID)
+			b.prevGset.(*MysqlGTIDSet).AddGTID(u, b.prevMySQLGTIDEvent.GNO)
```

Comment:
should we also consider the case "right after reconnect we will see same gtid as we saw before, thus currGset will not get changed"? (but I don't know when it will happen)

Reply:
That won't happen, because we will leave the currSet empty when we retry, and the mariadb comment will just stay as it is.
https://github.com/go-mysql-org/go-mysql/blob/7685eee9a345f53977d3c42e1aa220e0534f5cbf/replication/binlogsyncer.go#L664